server/index.js

```diff
@@ -69,20 +69,60 @@ app.get('/user/find', (req, res) => {
 	});
 });
 
+let waitingUser = {}
+
+const matchMaker = () => {
+	setInterval(() => {
+		if(userModule.getWaitingUserLen() > 1){
+			let keys = userModule.getWaitingUserKeys()
+			let roomval = userModule.getWaitingUserKeys()[0] + userModule.getWaitingUserKeys()[1]
+			userModule.getWaitingUser(keys[0]).join(roomval)
+			userModule.getWaitingUser(keys[1]).join(roomval)
+			io.to(roomval).emit('joined',"Searched completed")
+			userModule.getWaitingUser(keys[0])
+			let udata1 = {
+				id : keys[0],
+				room : roomval
+			}
+			let udata2 = {
+				id : keys[1],
+				room : roomval
+			}
+			userModule.addUser(udata1)
+			userModule.addUser(udata2)
+			userModule.addActiveUser({ id : keys[0] })
+			userModule.addActiveUser({ id : keys[1] })
+			userModule.delWaitingUser(keys[0])
+			userModule.delWaitingUser(keys[1])
+			console.log("running again n again",userModule.getWaitingUserLen(),keys)
+		}
+	},3000)
+}
+matchMaker()
 // Sockets
 io.on('connection', (socket) => {
-	socket.on('adding', (data) => {
-		if (data.userID.ID === '') return;
-		userModule.allUsers(data.userID.ID);
-	});
+	
+	socket.on('join',() => {
+		userModule.addWaitingUser(socket)
+	})
 
-	socket.on('createRoom', () => {
-		userModule.matchUsers(socket);
-	});
+	socket.on('pmessage',(message,callback) => {
```

Comment:
Specify pmessage means, doesn't matter how long the word is

Reply:
Fixed this made pmessage to private message.